Update README.md (#185)
digital-asset/cn-quickstart: README.md

# Canton Network application quickstart

> **Note**: On July 2nd, the quickstart had an architectural change and no longer connects to DevNet.  This update may require a `make clean-all` to remove old data and then a `make build`.
> 

Reviewer: ```suggestion
**Note**: On July 2, 2025, the quickstart underwent an architectural change and no longer connects to DevNet. This update may require a `make clean-all` to remove old data and then a `make build`.

```

This project provides scaffolding to develop a Canton Network application for the Global Synchronizer (CN GS). We intend that you clone the repository and incrementally update the solution to match your business operations. We assume that you have a Daml Enterprise license to leverage all of this project's features at runtime. However, an OSS developer can benefit from this project by understanding how a CN GS application is structured.

> To run the Quickstart you need some binaries from Artifactory.  Request Artifactory access by clicking [here](https://www2.digitalasset.com/contact-us-access-to-jfrog) and we will get right back to you.

Reviewer: ```suggestion
To run the Quickstart you need some binaries from Artifactory. Request Artifactory access by clicking [here](https://www2.digitalasset.com/contact-us-access-to-jfrog).
```

$ make shell
```

> When running `make start` for the first time, an assistant helps you set up the local deployment. You can choose to run the application in `DevNet` or `LocalNet` mode (recommended) for local development and testing; the latter meaning that a transient Super Validator is set up locally. You can change this later by running `make setup`.

Reviewer 2: @hrischuk-da @TheChronicMonster remove devnet from here

> When running `make start` for the first time, an assistant will help you setting up the local deployment. You can choose to run the application in `DevNet` or `LocalNet` mode (recommended) for local development and testing, the latter meaning that a transient Super Validator is set up locally. You can change this later by running `make setup`.

Reviewer: ```suggestion
When running `make start` for the first time, an assistant helps you set up the local deployment. You can choose to run the application in `DevNet` or `LocalNet` mode (recommended) for local development and testing; the latter meaning that a transient Super Validator is set up locally. You can change this later by running `make setup`.
```

## Debugging TL;DR

If a container fails to start, there are a few things to try:

- Ensure Docker Compose is configured to allocate enough memory. The recommended minimum total memory is 8 GB.
- Start fresh with `make clean-all` and then manually delete all Docker images and volumes.
- You may need to upgrade to a more recent version of the Daml SDK. Run `make install-daml-sdk` to assess your version and upgrade if you're not on the latest version.

**Note**: The CN QS uses Java SDK version `Eclipse Temurin JDK version 17.0.12+7` which runs within the Docker container.  This information is specified in `quickstart/compose.yaml` and `.env`.

> If you need assistance then the log information will be needed to debug the situation.  Please follow these directions to gather the logs:

Reviewer: ```suggestion
If you need assistance, please follow these directions to gather the log information needed for debugging:
```

This configuration demonstrates how the `backend-service` relies on the Quickstart-provided infrastructure. Quickstart automates much of the local environment setup for LocalNet, allowing you to prioritize application development. As you progress toward deployment and explore cloud orchestration, a deeper grasp of service configuration is invaluable. For now, consider these services a ready-to-use infrastructure foundation.

> Then explore `register-app-user-tenant`, the service that registers AppUser tenants to the `backend-service`. That allows end users from the AppUser organization to log in and Quickstart the web UI. That, in turn, ties the AppUser Identity Provider to the AppUser primary party ID. That means that if the end user is logged in through this Identity Provider, the user can then act as the AppUser primary party. The `register-app-user-tenant` service utilizes functionality provided by the `splice-onboarding` module to make the task as simple as possible.

Reviewer: ```suggestion
Then explore `register-app-user-tenant`, the service that registers AppUser tenants to the `backend-service`. This allows end users from the AppUser organization to log in and quickly start the web UI. That, in turn, ties the AppUser Identity Provider to the AppUser primary party ID. If the end user is logged in through this Identity Provider, the user can then act as the AppUser primary party. The `register-app-user-tenant` service utilizes functionality provided by the `splice-onboarding` module to make the task as simple as possible.
```
